app/src/main/java/ch/epfl/sdp/appart/user/UserViewModel.java

```diff
@@ -44,7 +45,11 @@ public UserViewModel(DatabaseService database, LoginService loginService) {
      */
     public void putUser(User user) {
         CompletableFuture<Boolean> putUser = db.putUser(user);
-        putUser.thenAccept(mPutCardConfirmed::setValue);
+        putUser.exceptionally(e -> {
+            Log.d("PUT USER", "DATABASE FAIL");
+            return null;
```

Comment:
Maybe add a way for the user to know there has been an error updating their info?
(for all the ops in this class)

Reply:
This is already the case at the moment. Each operation has a respective confirmation boolean : 
+ mPutUserConfirmed
+ mUpdateUserConfirmed
+ mUpdateImageConfirmed
+ mDeleteImageConfirmed

when one of these operations has finished in the UserProfileActivity will check whether the value of the above mentioned getters is true. If this is **not** the case the activity will show the error snackbar you commented above!